Sem. 04/Pract. 04/Task02.cpp

```diff
@@ -0,0 +1,202 @@
+#include <iostream>
+#include <fstream>
+
+#pragma warning( disable : 4996)
+
+const unsigned ARRAY_SIZE = 3;
+
+class Task {
+private:
+	char* name = nullptr;
+	char* description = nullptr;
+	unsigned points = 0;
+
+public:
+	Task() {
+		name = nullptr;
+		description = nullptr;
+		points = 0;
+	}
+	void setName(const char* name) {
+		if (this->name != nullptr) {
+			delete[] this->name;
+		}
+		
+		unsigned nameLen = 0;
+		nameLen = strlen(name);
+
+		this->name = new char[nameLen + 1];
+		strcpy(this->name, name);
+	}
+	void setNameFromFile(std::ifstream& ifs, unsigned nameLen) {
+		this->name = new char[nameLen+1];
+		ifs.read(this->name, nameLen);
+		this->name[nameLen] = '\0';
+	}
+
+	void setDescription(const char* description) {
+		if (this->description != nullptr) {
+			delete[] this->description;
+		}
+
+		unsigned nameLen = 0;
+		nameLen = strlen(description);
+
+		this->description = new char[nameLen + 1];
+		strcpy(this->description, description);
+		this->description[nameLen] = '\0';
+	}
+
+	void setDescriptionFromFile(std::ifstream& ifs, unsigned descriptionLen) {
+		this->description = new char[descriptionLen + 1];
+		ifs.read(this->name, descriptionLen);
```

Comment:
This causes the heap corruption. You are populating the wrong array. this->name has length less than this->description and when you read the description in the name array it overflows hence heap corruption